addons/crm/static/src/components/lead_generation_dropdown/lead_generation_dropdown.js

```diff
@@ -0,0 +1,272 @@
+import { Component, useState } from "@odoo/owl";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { useDropdownState } from "@web/core/dropdown/dropdown_hooks";
+import { useService } from "@web/core/utils/hooks";
+import { useHotkey } from "@web/core/hotkeys/hotkey_hook";
+import { sprintf } from "@web/core/utils/strings";
+import { _t } from "@web/core/l10n/translation";
+import { user } from "@web/core/user";
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+
+export const MODULE_STATUS = {
+    NOT_INSTALLED: "NOT_INSTALLED",
+    INSTALLING: "INSTALLING",
+    FAILED_TO_INSTALL: "FAILED_TO_INSTALL",
+    INSTALLED: "INSTALLED",
+};
+
+export class LeadGenerationDropdown extends Component {
+    static template = "crm.lead_generation_dropdown";
+    static props = {};
+    static components = { Dropdown, DropdownItem };
+
+    setup() {
+        this.orm = useService("orm");
+        this.dialogs = useService("dialog");
+        this.action = useService("action");
+        this.newContentText = {
+            FAILED_TO_INSTALL: _t('Failed to install "%s"'),
+            INSTALLING: _t('Installing "%s"'),
+            NOT_INSTALLED: _t('Do you want to install the "%s" App?'),
+        };
+
+        this.state = useState({
+            dropdownContentElements: [
+                {
+                    description: _t("Search in our directory of 300.000+ companies"),
+                    hasAccess: user.isAdmin,
+                    icon: "/crm/static/src/img/lead_sourcing.png",
+                    index: 10,
```

Comment:
Would call it "sequence", just because that's usually the terminology we use for this